terraform/modules/cloud/aws/gateway/main.tf

```diff
@@ -1,3 +1,75 @@
+resource "aws_acm_certificate" "cert" {
+  domain_name       = "${var.api_dns}"
+  validation_method = "DNS"
+}
+
+resource "aws_route53_record" "cert_validation" {
+  name    = "${aws_acm_certificate.cert.domain_validation_options.0.resource_record_name}"
+  type    = "${aws_acm_certificate.cert.domain_validation_options.0.resource_record_type}"
+  zone_id = "${var.dns_zone}"
+  records = ["${aws_acm_certificate.cert.domain_validation_options.0.resource_record_value}"]
+  ttl     = 60
+}
+
+resource "aws_route53_record" "main-api" {
+  zone_id = "${var.dns_zone}"
+  name    = "${var.api_dns}"
+  type    = "A"
+
+  alias {
+    name                   = "${aws_cloudfront_distribution.cf.domain_name}"
+    zone_id                = "${aws_cloudfront_distribution.cf.hosted_zone_id}"
+    evaluate_target_health = false
+  }
+}
+
+resource "aws_cloudfront_distribution" "cf" {
+  enabled = true
+
+  origin {
+    domain_name = "origin-${var.api_dns}"
+    origin_id   = "main"
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "http-only"
+      origin_ssl_protocols   = ["TLSv1.2"]
+    }
+  }
+
+  default_cache_behavior {
+    allowed_methods = ["HEAD", "DELETE", "POST", "GET", "OPTIONS", "PUT", "PATCH"]
+    cached_methods  = ["GET", "HEAD"]
+
```

Comment:
We should disable the cache for now, as the backend does not provide any correct headers.

e.g.
```
$ curl https://api.mainnet.ops.aeternity.com/v2/blocks/top
{"key_block":{"beneficiary":"ak_nv5B93FPzRHrGNmMdTDfGdd5xGZvep3MVSpJqzcQmMp59bBCv","hash":"kh_FK9GA6GCEA86rKsNiBbYyJcVwxTY2FGK23oXBp6YrWLXSnvds","height":61629,"info":"cb_Xfbg4g==","miner":"ak_hnDcvx4d4iDeu7rfy6Qqzvi7o2fSVaqi6Gmr9vLa77RT2ndXN","nonce":4500787326943769424,"pow":[11037633,14049166,18351931,44538717,48379180,80663004,93837005,117760316,122409925,148584423,163570036,188161604,189447809,192222538,197896877,208726636,227733434,255027211,271188957,280516759,285683009,304919303,319366757,330383191,330640805,331012890,333423107,335948363,376446540,380022069,390581760,408617207,423394541,442156706,451531256,452252749,456808955,486727191,490853354,491510639,500386774,532620813],"prev_hash":"kh_2EQtRvWid8VE7vaBDqQcMYrZnhfQJMDQx2TrpRHfTCLVrU3R9r","prev_key_hash":"kh_2EQtRvWid8VE7vaBDqQcMYrZnhfQJMDQx2TrpRHfTCLVrU3R9r","state_hash":"bs_2ktRzt7PWSBWBfUri81nQpZMfGFBwWxV9ezcJj2kNL52cJwJrv","target":503604072,"time":1554458999643,"version":2}}
dincho at Dinchos-Mac-mini in ~/Projects/aeternity/aeternity on master [?$]
$ curl http://3.0.12.164:3013/v2/blocks/top
{"key_block":{"beneficiary":"ak_nv5B93FPzRHrGNmMdTDfGdd5xGZvep3MVSpJqzcQmMp59bBCv","hash":"kh_PFhXyytJc8nTU7z1Bh2UgZaN5a3eRZxoaJzQxgDy1SXKY6G2h","height":61636,"info":"cb_Xfbg4g==","miner":"ak_q2sQ6AviNpDCCgA2DPzeJNyiLabneyYq5tHiaJMeiHRSf8Kz5","nonce":9193306283289704931,"pow":[6668953,35920853,54383087,58162659,61901876,91855754,129665421,130957220,131776393,157178801,176075634,188097206,203497605,203622723,207590997,207616137,241128535,259067311,278468295,281159164,283678169,284687186,296505241,317476757,319768706,327580160,342473808,342676643,355939080,372224130,407096455,410137476,415154591,425526145,434457538,438060351,447978415,451045908,460116667,487243155,499893167,527173883],"prev_hash":"kh_BcLnCYFQvg1U1Gm8S5cvneLxsqqnDwuhZDEmySZNgRXCGiyqx","prev_key_hash":"kh_BcLnCYFQvg1U1Gm8S5cvneLxsqqnDwuhZDEmySZNgRXCGiyqx","state_hash":"bs_WYVfpV7rmxMeBJCKJBrEozg6KFxMW96NuiaaRMpZfddiKRhfM","target":503626179,"time":1554460072674,"version":2}}

```

So `default_ttl = 0`. I guess you have to manually invalidate the cache after you change this.

While on it, perhaps enable a compression ? `compress = 1` ?